Modules/_json.c

```diff
@@ -1535,84 +1592,35 @@ encoder_listencode_dict(PyEncoderObject *s, _PyUnicodeWriter *writer,
         */
     }
 
-    items = PyMapping_Items(dct);
-    if (items == NULL)
-        goto bail;
-    if (s->sort_keys && PyList_Sort(items) < 0) {
-        Py_DECREF(items);
-        goto bail;
-    }
-    it = PyObject_GetIter(items);
-    Py_DECREF(items);
-    if (it == NULL)
-        goto bail;
-    idx = 0;
-    while ((item = PyIter_Next(it)) != NULL) {
-        PyObject *encoded, *key, *value;
-        if (!PyTuple_Check(item) || PyTuple_GET_SIZE(item) != 2) {
-            PyErr_SetString(PyExc_ValueError, "items must return 2-tuples");
+    if (s->sort_keys) {
+        Py_ssize_t i;
+
+        items = PyMapping_Items(dct);
```

Comment:
```suggestion
        items = PyDict_Items(dct);
```

Hmm, I would like to use `PyDict_Items` if possible, but I am checking the possibility of regression if the object that supports Mapping Protocol, not a subclass of dict. (I am not sure it's possible)

Reply:
https://github.com/python/cpython/blob/8ae1d21876974ed9c1a472f6132f47e51310c3ca/Modules/_json.c#L1439 

Okay let's use `PyDict_Items`

Reply:
There is also `PyDict_GET_SIZE` https://github.com/python/cpython/pull/95374/files#diff-efe183ae0b85e5b8d9bbbc588452dd4de80b39fd5c5174ee499ba554217a39edL1506 which has an assert on `PyDict_Check`. So at least the debug version ensures it's always a dict.